dds/DCPS/transport/framework/TransportInst.cpp

```diff
@@ -124,4 +124,27 @@ OpenDDS::DCPS::TransportInst::impl()
   return this->impl_;
 }
 
+void
+OpenDDS::DCPS::TransportInst::set_port_in_addr_string(OPENDDS_STRING& addr_str, u_short port_number)
+{
+  const int BUFSIZE=1024;
+  char result[BUFSIZE];
+
+  if (std::count(addr_str.begin(), addr_str.end(), ':') < 2) {
```

Comment:
OK, in my earlier review didn't see that the -1 value for 'pos' will be ignored by snprintf